### DISCUSSION TITLE
Problema ad aggiungere Sonoff flashato Tasmota ad Homeassistant

### DISCUSSION THREAD

**Discussion by lnrd00** [indomus/forum#2990] [Hardware & Firmware]
Salve, avrei una domanda.
Ho flashato un Sonoff TH16 con firmware Tasmota, versione 13.4.0, e riesco a controllarla dalla sua web API inserendo l'indirizzo IP adeguato.
Il problema è che non riesco ad aggiungerla a Homeassistant attraverso l'integrazione Tasmota, che ho già installato su Homeassistant e ho anche già un altro Sonoff connesso (un Sonoff Basic R3). 
Sbaglio forse nel fare la configurazione MQTT? Nella schermata di controllo che vedo attraverso la web API mi sembrava di star configurando tutto correttamente:
<img width="377" alt="Schermata 2024-03-20 alle 19 56 54" src="https://github.com/indomus/forum/assets/154708323/39254b77-ca40-4925-ba66-ce9e956477fb">
Qualcuno saprebbe dirmi quale potrebbe essere il problema? Grazie in anticipo!
Quando dico che "non riesco ad aggiungerla" intendo dire che Tasmota non si accorge, e dunque non aggiunge automaticamente, il dispositivo flashato.

**Comment by snakuzzo** [comment]
Assicurati dalla che il `SetOption19` sia a `OFF`.
Dalla webgui del device clicca su "Console" e digita... `SO19`

Dovresti avere un risultato simile...

```
21:45:48.390 CMD: SO19
21:45:48.398 MQT: stat/tasmota-forno/RESULT = {"SetOption19":"OFF"}
```

Se dovesse essere `"ON"` digita...

```
SO19 0
```

A quel punto dovrebbe essere `"OFF"` e dovrebbe cominciare a pubblicare i messaggi per l'autodiscovery dell'integrazione tasmota...e di conseguenza dovresti poter aggiungere il device con integrazione tasmota.

Comunque è tutto scritto sulla pagina ufficiale dell'integrazione...

si su HA -> [QUI](https://www.home-assistant.io/integrations/tasmota/)
che sul progetto Tasmota -> [QUI](https://tasmota.github.io/docs/Home-Assistant/)

**Reply by KC-inDomus** [reply]
aggiungo: hai l'autodiscovery mqtt attivo?

**Reply by snakuzzo** [reply]
Non usa autodiscovery mqtt, ma autodiscovery dell'integrazione tasmota. Ha già un altro device con la stessa integrazione.

**Reply by lnrd00** [reply]
Purtroppo, `SetOption19` sembrerebbe essere già stata su `OFF`.
Questi sono i log presenti in console:
<img width="1271" alt="Schermata 2024-03-21 alle 16 31 59" src="https://github.com/indomus/forum/assets/154708323/bf864e0a-f58f-4325-99fd-9518058003fd">

**Reply by KC-inDomus** [reply]
leggi il log: connessione fallita verso "`ip-del-broker-mqtt`" (...)

**Reply by snakuzzo** [reply]
`ip-del-broker-mqtt` 

OTTIMO!!! :D

,

**Reply by lnrd00** [reply]
Non ricordo cos'è che dovevo inserire al posto di `ip-del-broker-mqtt`🥲
Forse era `mqtt://127.0.0.1` ? Perché se così fosse comunque non funziona dato che ho già provato...

**Reply by snakuzzo** [reply]
Ma come pretendi che funzioni 127.0.0.1 che sarebbe localhost del device?
Devi mettere l'IP LAN del server su cui gira il broker mqtt. Potresti banalmente copiare quello che hai messo sull'altro device tasmota che funziona...non trovi?

**Reply by lnrd00** [reply]
Ho deciso di scrivere sul forum dopo aver provato diverse volte ad inserire l'IP LAN del Raspberry...quindi non capisco come mai questa volta abbia funzionato🫤grazie mille e scusate per la perdita di tempo!
